wardrobe/tradable_kitties/src/tests.rs

```diff
@@ -0,0 +1,928 @@
+    let new_family = TradableKittyData::default_family();
+    let mut new_child = new_family[2].clone();
+    new_child.kitty_basic_data.dna = KittyDNA(H256::zero());
+
+    let result = TradableKittyConstraintChecker::<0>::Breed.check(
+        &[
+            TradableKittyData::default().into(),
+            TradableKittyData::default_dad().into(),
+        ],
+        &[], // no peeks
+        &[
+            new_family[0].clone().into(),
+            new_family[1].clone().into(),
+            new_child.into(),
+        ],
+    );
+    assert_eq!(
+        result,
+        Err(TradableKittyConstraintCheckerError::NewChildDnaIncorrect)
+    );
+}
+
+#[test]
+fn check_child_dad_parent_tired_fails() {
+    let new_family = TradableKittyData::default_family();
+    let mut new_child = new_family[2].clone();
+    new_child.kitty_basic_data.parent = Parent::Dad(DadKittyStatus::Tired);
+
+    let result = TradableKittyConstraintChecker::<0>::Breed.check(
+        &[
+            TradableKittyData::default().into(),
+            TradableKittyData::default_dad().into(),
+        ],
+        &[], // no peeks
+        &[
+            new_family[0].clone().into(),
+            new_family[1].clone().into(),
+            new_child.into(),
+        ],
+    );
+    assert_eq!(
+        result,
+        Err(TradableKittyConstraintCheckerError::NewChildIncorrectParentInfo)
+    );
+}
+
+#[test]
+fn check_child_mom_parent_recently_gave_birth_fails() {
+    let new_family = TradableKittyData::default_family();
+    let mut new_child = new_family[2].clone();
+    new_child.kitty_basic_data.parent = Parent::Mom(MomKittyStatus::HadBirthRecently);
+
+    let result = TradableKittyConstraintChecker::<0>::Breed.check(
+        &[
+            TradableKittyData::default().into(),
+            TradableKittyData::default_dad().into(),
+        ],
+        &[], // no peeks
+        &[
+            new_family[0].clone().into(),
+            new_family[1].clone().into(),
+            new_child.into(),
+        ],
+    );
+    assert_eq!(
+        result,
+        Err(TradableKittyConstraintCheckerError::NewChildIncorrectParentInfo)
+    );
+}
+
+#[test]
+fn check_child_free_breedings_incorrect_fails() {
+    let new_family = TradableKittyData::default_family();
+    let mut new_child = new_family[2].clone();
+    new_child.kitty_basic_data.free_breedings = KittyHelpers::NUM_FREE_BREEDINGS + 1;
+
+    let result = TradableKittyConstraintChecker::<0>::Breed.check(
+        &[
+            TradableKittyData::default().into(),
+            TradableKittyData::default_dad().into(),
+        ],
+        &[], // no peeks
+        &[
+            new_family[0].clone().into(),
+            new_family[1].clone().into(),
+            new_child.into(),
+        ],
+    );
+    assert_eq!(
+        result,
+        Err(TradableKittyConstraintCheckerError::NewChildFreeBreedingsIncorrect)
+    );
+}
+
+#[test]
+fn check_child_num_breedings_non_zero_fails() {
```

Comment:
I guess my only issues here is that we are just repeating alot of the same logic from the kitties piece. I think is somewhat silly if we cant find a way to not repeat ourselfs each time we want composable pieces..